integration/simulation/simulation.go

```diff
@@ -44,6 +44,7 @@ func NewSimulation(
 	gossipPeriod uint64,
 	localEnclave bool,
 	stats *Stats,
+	newP2P func(ourAddress string, allAddresses []string) p2p.P2P,
```

Comment:
I see this as a parameter in the L2NetworkConfig object

Reply:
can't this better be an interface?

Reply:
Was trying to avoid abstractions. But it is not pretty. Create a factory for injecting the p2p type while allowing the instantiation of the object to be further down the execution path.